### ISSUE TITLE
False positive for `Naming/MemoizedInstanceVariableName` and `Lint/Syntax` cops

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#13434]
## Actual behavior

The code:
```ruby
class Temp
  def banners_only=(value)
    @banners_only = value

    @banners_only ||= allowed_for_banners_only(allowed_banners)
  end
end
```

The rubocop check:
```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/temp.rb
Inspecting 1 file
F

Offenses:

app/models/temp.rb:7:5: C: [Correctable] Naming/MemoizedInstanceVariableName: Memoized variable @banners_only does not match method name banners_only=. Use @banners_only= instead.
    @banners_only ||= allowed_for_banners_only(allowed_banners)
    ^^^^^^^^^^^^^
app/models/temp.rb:7:20: F: Lint/Syntax: unexpected token tOP_ASGN
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
    @banners_only ||= allowed_for_banners_only(allowed_banners)
                   ^^^

1 file inspected, 2 offenses detected, 1 offense autocorrectable
```

## RuboCop version

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.68.0 (using Parser 3.3.4.2, rubocop-ast 1.32.2, analyzing as Ruby 2.7, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-performance 1.22.1
  - rubocop-rails 2.27.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.2.0
  - rubocop-rspec_rails 2.30.0
  - rubocop-thread_safety 0.5.1
```
